controllers/resource/upserter_test.go

```diff
@@ -489,6 +560,40 @@ func Test_upserter_UpsertDoguPVCs(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, expectedDoguPVC, actualDoguPVC)
 	})
+
+	t.Run("success when upserting a new dogu pvc when an old pvc is terminating", func(t *testing.T) {
+		// given
+		doguResource := readLdapDoguResource(t)
+		dogu := readLdapDogu(t)
+		var reservedPvc *v1.PersistentVolumeClaim
+		now := metav1.Now()
+		reservedPvc = readLdapDoguExpectedDoguPVC(t)
+		reservedPvc.DeletionTimestamp = &now
+		client := fake.NewClientBuilder().WithScheme(getTestScheme()).WithObjects(doguResource, reservedPvc).Build()
+		timer := time.NewTimer(time.Second * 5)
+		go func() {
+			<-timer.C
+			err := client.Delete(context.Background(), reservedPvc)
+			require.NoError(t, err)
+		}()
+
+		generator := mocks.NewDoguResourceGenerator(t)
+		expectedDoguPVC := readLdapDoguExpectedDoguPVC(t)
+		generator.On("CreateDoguPVC", doguResource).Return(expectedDoguPVC, nil)
```

Comment:
Just a heads-up which I recently learned from @jelemux: 
Mockery is now capable of doing Java-style proxy object mocking. When properly generated, the function being mocked no longer needs to be addressed by string but instead like so:
```golang
generator.EXPECT().CreateDoguPVC(doguResource).Return(expected, nil)
```

In my eyes a massive drop of convenience :heart_eyes: 

There is a makefile `mocks.mk` in the makefile repo starting with [v7.3.0](https://github.com/cloudogu/makefiles/releases/tag/v7.3.0). See also more here https://vektra.github.io/mockery/configuration/